runtime/ft_test.go

```diff
@@ -766,3 +770,408 @@ func BenchmarkRuntimeFungibleTokenTransfer(b *testing.B) {
+			return parser.ParseProgram(memoryGauge, []byte(code), parser.Config{})
+		},
+	}
+
+	environment := NewBaseInterpreterEnvironment(Config{})
+
+	nextTransactionLocation := NewTransactionLocationGenerator()
+
+	// Deploy Fungible Token contract
+
+	err := runtime.ExecuteTransaction(
+		Script{
+			Source: utils.DeploymentTransaction(
+				"FungibleToken",
+				[]byte(modifiedFungibleTokenContractInterface),
+			),
+		},
+		Context{
+			Interface:   runtimeInterface,
+			Location:    nextTransactionLocation(),
+			Environment: environment,
+		},
+	)
+	require.NoError(t, err)
+
+	// We cannot deploy the ExampleToken contract because it is broken.
+	// Manually storage the contract value for the ExampleToken contract in the contract account's storage
+
+	storage, inter, err := runtime.Storage(Context{
+		Interface: runtimeInterface,
+	})
+	require.NoError(t, err)
+
+	contractValue := interpreter.NewCompositeValue(
+		inter,
+		interpreter.EmptyLocationRange,
+		common.NewAddressLocation(nil, contractsAddress, contractName),
+		contractName,
+		common.CompositeKindContract,
+		[]interpreter.CompositeField{
+			{
+				Name: fungibleTokenTypeTotalSupplyFieldName,
+				Value: interpreter.NewUnmeteredUFix64ValueWithInteger(
+					4321,
+					interpreter.EmptyLocationRange,
+				),
+			},
+		},
+		contractsAddress,
+	)
+
+	contractStorage := storage.GetStorageMap(
+		contractsAddress,
+		StorageDomainContract,
+		true,
+	)
+	contractStorage.SetValue(
+		inter,
+		interpreter.StringStorageMapKey(contractName),
+		contractValue,
+	)
+
+	// Manually store a broken ExampleToken.Vault in the user account's storage
+
+	vaultValue := interpreter.NewCompositeValue(
+		inter,
+		interpreter.EmptyLocationRange,
+		common.NewAddressLocation(nil, contractsAddress, contractName),
+		fmt.Sprintf("%s.Vault", contractName),
+		common.CompositeKindResource,
+		[]interpreter.CompositeField{
+			{
+				Name:  sema.ResourceUUIDFieldName,
+				Value: interpreter.NewUnmeteredUInt64Value(42),
+			},
+			{
+				Name: fungibleTokenVaultTypeBalanceFieldName,
+				Value: interpreter.NewUnmeteredUFix64ValueWithInteger(
+					1234,
+					interpreter.EmptyLocationRange,
+				),
+			},
+		},
+		userAddress,
+	)
+
+	userStorage := storage.GetStorageMap(
+		userAddress,
+		common.PathDomainStorage.Identifier(),
+		true,
+	)
+	const storagePathIdentifier = "exampleTokenVault"
+	userStorage.SetValue(
+		inter,
+		interpreter.StringStorageMapKey(storagePathIdentifier),
+		vaultValue,
+	)
+
+	err = storage.Commit(inter, false)
+	require.NoError(t, err)
+
+	// Send a transaction that loads the broken ExampleToken contract and the broken ExampleToken.Vault
+
+	const transaction = `
+      import FungibleToken from 0x1
+      import ExampleToken from 0x1
+
+      transaction {
+
+          let vault: @ExampleToken.Vault
+
+          prepare(signer: auth(LoadValue) &Account) {
+              self.vault <- signer.storage.load<@ExampleToken.Vault>(from: /storage/exampleTokenVault)!
```

Comment:
Could we add some more operations to the test? e.g:
- getting the type (`self.vault.getType()`)
- casting to `FungibleToken.Vault`
- Storing the read value back to storage
- Calling a function on the `self.vault`
- etc.

Reply:
Good idea! Will add these 👍

Reply:
Added in [7197c69](https://github.com/onflow/cadence/pull/3482/commits/7197c6908705c98e8cc791c342efb2da6cfe9c72) and [acd782a](https://github.com/onflow/cadence/pull/3482/commits/acd782a9a053181c46194063913a0bbb81f46abb)